bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptEngineFactory.java

```diff
@@ -16,44 +16,49 @@
 import java.util.Map;
 
 import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+
+import org.openhab.core.automation.module.script.internal.provider.ScriptModuleTypeProvider;
 
 /**
- * This class is used by the ScriptManager to load ScriptEngines.
- * This is meant as a way to allow other OSGi bundles to provide custom Script-Languages with special needs (like
- * Nashorn, Groovy, etc.)
+ * This class is used by the ScriptEngineManager to load ScriptEngines. This is meant as a way to allow other OSGi
+ * bundles to provide custom script-languages with special needs, e.g. Nashorn, Jython, Groovy, etc.
  *
  * @author Simon Merschjohann
- *
+ * @author Scott Rushworth - added/changed methods and parameters when implementing {@link ScriptModuleTypeProvider}
  */
 public interface ScriptEngineFactory {
 
+    final static ScriptEngineManager engineManager = new ScriptEngineManager();
+
     /**
-     * @return the list of supported language endings e.g. py, jy
+     * This method logs details about the implementations of ScriptEngineFactory.
+     *
      */
-    List<String> getLanguages();
+    void logFactoryDetails();
 
     /**
-     * "scopes" new values into the given ScriptEngine
+     * This method returns a list of file extensions and MimeTypes that are supported by the ScriptEngine, e.g. py,
+     * application/python, js, application/javascript, etc.
      *
-     * @param scriptEngine
-     * @param scopeValues
+     * @return List of supported script types
      */
-    void scopeValues(ScriptEngine scriptEngine, Map<String, Object> scopeValues);
+    List<String> getScriptTypes();
 
     /**
-     * created a new ScriptEngine
+     * This method "scopes" new values into the given ScriptEngine.
      *
-     * @param fileExtension
-     * @return
+     * @param scriptEngine
+     * @param scopeValues
      */
-    ScriptEngine createScriptEngine(String fileExtension);
+    void scopeValues(ScriptEngine scriptEngine, Map<String, Object> scopeValues);
 
     /**
-     * checks if the script is supported. Does not necessarily be equal to getLanguages()
+     * This method creates a new ScriptEngine based on the supplied file extension or MimeType.
```

Comment:
The JavaDoc should declare how a invalid / unknown `scriptType` is handled.
Is `null` returned or is an `IllegalArgumentException` thrown?

If `null` is returned, it would make sense to add `@NonNullByDefault` to the class and add `@Nullable`

Reply:
This was an existing method that I renamed and slightly modified, but I understand the need for better documentation. 

Worst case, [javax.script.ScriptEngineManager.getEngineFactories()](https://docs.oracle.com/javase/8/docs/api/javax/script/ScriptEngineManager.html#getEngineFactories--) will return an empty list and does not throw anything, as does [javax.script.ScriptEngineFactory.getExtensions()](https://docs.oracle.com/javase/8/docs/api/javax/script/ScriptEngineFactory.html#getExtensions--) and [javax.script.ScriptEngineFactory.getMimeTypes()](https://docs.oracle.com/javase/8/docs/api/javax/script/ScriptEngineFactory.html#getMimeTypes--), causing getScriptTypes() to return an empty list. If there is no ScriptEngineManager, there are some serious problems and I can't imagine we'd get far enough to call this method. So, this looks to me like getScriptTypes() would never return null.

Reply:
`Collections.emptyList()` :wink: